### ISSUE TITLE
build: strip the produced cockroach-sql binary

### ISSUE THREAD

**Issue by knz** [cockroachdb/cockroach#79675]
Followup to #75800; the `cockroach-sql` binary should be stripped to ensure it remains small.

Epic: CC-5993

Jira issue: CRDB-14975

**Comment by rail** [comment]
Thank you for filing this!

**Comment by kenliu-crl** [comment]
manually reviewed and updated

**Comment by ajwerner** [comment]
We should use https://upx.github.io/. Stripping will only affect the DWARF debug symbols and won't compress other data in the binary like go's pclntab.

**Comment by knz** [comment]
correct, but the DWARF symbol table is 20-40% of go binary size and completely redundant with go's internal structures so definitely important to strip.

**Comment by rail** [comment]
back to the pool for now
